packages/components/managed-data-table/src/data-table-settings/column-manager/column-manager.js

```diff
@@ -0,0 +1,231 @@
+import debounce from 'debounce-promise';
+import differenceWith from 'lodash.differencewith';
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import requiredIf from 'react-required-if';
+import { useIntl } from 'react-intl';
+import { DragDropContext } from 'react-beautiful-dnd';
+import AsyncSelectInput from '@commercetools-uikit/async-select-input';
+import CollapsibleMotion from '@commercetools-uikit/collapsible-motion';
+import FieldLabel from '@commercetools-uikit/field-label';
+import Spacings from '@commercetools-uikit/spacings';
+import Text from '@commercetools-uikit/text';
+import {
+  CloseIcon,
+  EyeCrossedIcon,
+  EyeIcon,
+  SearchIcon,
+} from '@commercetools-uikit/icons';
+import AccessibleButton from '@commercetools-uikit/accessible-button';
+import { filterDataAttributes } from '@commercetools-uikit/utils';
+import DroppablePanel from '../droppable-panel';
+import {
+  Container,
+  HeaderContainer,
+  DroppableContainer,
+} from '../data-table-settings-panel.styles';
+import messages from './messages';
+
+export const ColumnManager = (props) => {
+  const { formatMessage } = useIntl();
+  const [isDragging, setIsDragging] = useState(false);
+  const handleDragEnd = (result) => {
+    setIsDragging(false);
+
+    // Invalid drop destination, do nothing
+    if (!result.destination) return;
+    if (result.destination.droppableId === 'hidden-columns-panel') {
+      if (result.source.droppableId === 'hidden-columns-panel') return;
```

Comment:
I might be missing something but this check looks odd to me.

Reply:
When the destination is the hidden columns panel the origin can be:
- (also) the _hidden columns panel_: no change is done (as makes no sense to alter the order of the columns there)
- the _selected columns panel_: the column should be added to the panel.

We can do first a 
```
if (result.destination.droppableId === 'hidden-columns-panel' 
    && result.source.droppableId === 'hidden-columns-panel') return;
```

But then you would need to validate again the destination... I think the logic makes sense

Reply:
Ah crap. Didn't see it's source and destination 🤕. Should we have a `getIsDesinationColumnsPanel` and `getIsSourceColumnsPanel`?

Reply:
| condition | number of comparisons |
| --- | --- |
| `result.destination.droppableId === HIDDEN_COLUMNS_PANEL` |  1 |
| `result.source.droppableId === HIDDEN_COLUMNS_PANEL` | 1  |
| `result.source.droppableId === SELECTED_COLUMNS_PANEL` | 1 |

We would have:
- `getIsDestinationColumnsPanel(result, HIDDEN_COLUMNS_PANEL)` called 1 time
- `getIsSourceColumnsPanel(result, HIDDEN_COLUMNS_PANEL)` called 1 time
- `getIsSourceColumnsPanel(result, SELECTED_COLUMNS_PANEL)` called 1 time

I don't mind adding it for readability, just stating the result